controllers/add_process_groups.go

```diff
@@ -0,0 +1,112 @@
+/*
+ * add_process_groups.go
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2021 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import (
+	ctx "context"
+	"fmt"
+	"time"
+
+	fdbtypes "github.com/FoundationDB/fdb-kubernetes-operator/api/v1beta1"
+)
+
+// AddProcessGroups provides a reconciliation step for adding new pods to a cluster.
+type AddProcessGroups struct{}
+
+// Reconcile runs the reconciler's work.
+func (a AddProcessGroups) Reconcile(r *FoundationDBClusterReconciler, context ctx.Context, cluster *fdbtypes.FoundationDBCluster) (bool, error) {
+	desiredCountStruct, err := cluster.GetProcessCountsWithDefaults()
+	if err != nil {
+		return false, err
+	}
+	desiredCounts := desiredCountStruct.Map()
+
+	processCounts := make(map[string]int)
+	processGroupIDs := make(map[string]map[int]bool)
+	for _, processGroup := range cluster.Status.ProcessGroups {
+
+		processGroupID := processGroup.ProcessGroupID
+		_, num, err := ParseInstanceID(processGroupID)
```

Comment:
```suggestion
		_, id, err := ParseInstanceID(processGroupID)
```

Reply:
What's the advantage of changing this to "id"? We generally refer to it as a number in other uses.

Reply:
Right it's only the process "number" and not the whole ID